accepted/00070-akka.md

```diff
@@ -0,0 +1,141 @@
+ - most but not all Messages were made `Serializable`, more refactoring might be needed to ensure this is always the case
+ - Akka tuning knobs were not all exposed, or exposed by reusing existing configuration variables. New variables might need to be introduced and properly documented
+ - ideas from [the normalisation PR](https://github.com/uyuni-project/uyuni/pull/1954) should probably be integrated
+
+5. move specific Taskomatic task facilities (eg. `ErrataCacheDriver`, `SSHPushDriver`) to Akka
+6. refactor Taskomatic itself to use Akka
+
+Last point is about having Quartz-based Taskomatic logic only caring aobut the triggering of jobs, and leave to Akka their execution. Basically when Taskomatic fires (eg. on cron line) it will send a message to an Actor which would then carry out the actual computation.
+
+## Following Steps
+
+Following points need further research when points 1.-6. are completed, and possibly need follow-up RFCs, still are reported here for completeness.
+
+### Move Taskomatic code in the Tomcat web app
+
+After all asynchronous computation is moved to Akka, it might make sense to consider moving all Taskomatic code out of the currently separated daemon snd into the main Tomcat web app.
+
+Architecturally, such a change would have pros and cons:
+ - pro: we could remove the internal Taskomatic XMLRPC API, eliminating substantial complexity and the synchronization/availability problems we currently face (eg. [the race condition in `MinionActionExecutor`](https://github.com/uyuni-project/uyuni/blob/4.1-alpha1/java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java#L82-L102), as per [downstream issue 10657](https://github.com/SUSE/spacewalk/issues/10657))
+   - it can be expected most of the methods can be removed - possibly all
+   - if some really need to be retained then it is a possibility to keep the endpoint on its current (Taskomatic-specific) TCP port or to map the methods to the Tomcat XMLRPC TCP port and adapt scripts using them 
+ - pro: with one DB connection pool and one Akka thread pool, it should result in an easier-to-tune Server
+ - pro: currently many support issues boil down to undetected Taskomatic failures. With one only web app, it should be easier to surface errors in a more user-visible way
+ - con: unexpected Taskomatic errors may bring Tomcat down as well. At the moment this does not happen, but as we rely on Taskomatic much more than in original Spacewalk, in many cases we already end up with a non-functioning Server anyway
+ - pro: currently Taskomatic runs as `root`, but not all tasks really require `root` privileges. Attack surface can thus be reduced by running code with Tomcat privileges by default
+ - con: Taskomatic tasks that do require root privileges will need some different permissions to run. One way is to use Akka Clustering (described below) with two processes (one `root` and one non-`root`) on the same machine
```

Comment:
Why is root required? I suppose for opening a well-know port (< 1024). Is my assumption correct?

Reply:
Not really, opening ports is not the main concern.

Problem here is that some tasks require `root` to be executed successfully, for example in order to invoke `spacewalk-repo-sync`.

Reply:
> Not really, opening ports is not the main concern.
> 
> Problem here is that some tasks require `root` to be executed successfully, for example in order to invoke `spacewalk-repo-sync`.

Why does `spacewalk-repo-sync` require root in the first place?

After task 1-5 are complete, some research would be needed to identify which actions Taskomatic currently runs as root, why those need root, and whether we can make them not need root.